File: js/cm-api-v2/PostItem.js
// DocSection: cm_api_v2_post_item
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
// Using ES6 syntax
import { ManagementClient } from '@kentico/kontent-management';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

client.addContentItem()
  .withData(
    {
      name: 'On Roasts',
      codename: 'my_article_my_article',
      type: {
        codename: 'article'
      },
      external_id: '59713'
    }
  )
  .toObservable()
  .subscribe((response) => {
    console.log(response);
  },
  (error) => {
    console.log(error);
  });
// EndDocSection

File: net/cm-api-v2/PostItem.cs
// DocSection: cm_api_v2_post_item
// Tip: Find more about .NET SDKs at https://docs.kontent.ai/net
using Kentico.Kontent.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_PROJECT_ID>"
});

var response = await client.CreateContentItemAsync(new ContentItemCreateModel
{
    Name = "On Roasts",
    Codename = "my_article",
    Type = Reference.ByCodename("article"),
    Collection = Reference.ByCodename("default"),
    ExternalId = "59713",
});
// EndDocSection
